src/tests/test_model.py

```diff
@@ -71,13 +83,46 @@ def test_model_vit(datapipe):
         )
         trainer.fit(model=model, train_dataloaders=dataloader)
 
+
+@pytest.mark.parametrize(
+    "litmodule,precision",
+    [
+        (CLAYModule, "bf16-mixed" if torch.cuda.is_available() else "32-true"),
```

Comment:
There are some `torch.cat` calls in CLAYModule that don't work when run on CPU with float16 tensors, see https://github.com/pytorch/pytorch/issues/100932. The patch at https://github.com/pytorch/pytorch/issues/96093 to fix this issue is merged already though, so we can remove this if-then statement in the future when Pytorch 2.2 is out. Note that running CLAYModule on CUDA-enabled GPUs should be fine with float16 or bfloat16.